es_loader/es_loader.py

```diff
@@ -41,12 +72,49 @@ def _create_index(self):
         self.es.indices.create(self.index_name)
         logger.debug('Index created')
 
+    def _get_job(self):
+        """Get job from ScrapingHubClient istance."""
+        self.job = self.sc.get_job(self.job_id)
+
+    def _get_job_metadata(self):
+        """Get job metadata."""
+        self.metadata = self.job.metadata.list()
+
+    def _get_scraped_items_count(self):
+        """Get scraped items count from job metadata.
+
+        The count is needed to make log messages more verbose
+        and to calculate batches size"""
+        self._get_job_metadata()
+        for data in self.metadata:
+            if data[0] == 'scrapystats':
+                self.items_count = data[1]['item_scraped_count']
+
     def _get_items(self):
-        logger.debug('Getting items')
-        items = self.sc.get_job(self.job_id).items.iter()
+        """Get items from Scrapy Cloud.
+
+        Returns:
+            items - an iterator with items.
+        """
+        logger.debug(
+            'There are {} items scraped. Downloading...'.format(self.items_count)
+        )
+        items = self.job.items.iter()
         return items
 
     def _index_item(self, item):
+        """Prepare item to be indexed to ElasticSearch.
+
+        This method wrapps item from Scrapy Cloud to the be ready
+        to be indexed into ElasticSearch.
+        Also here actual batching is happening - items are accumulated
+        into buffer and then sent to ElasticSearh in bulk.
+
+        Arguments:
+            item - an item from items() iterator of ScrapingHubClient.Job
+        """
+        if self.buffer_size == 0:
+            return
```

Comment:
Does it make sense to have this check, considering that buffer_size is hardcoded in the constructor? 
Also, even if it weren't hardcoded, I'd better raise an exception instead of silently exiting and making the user wonder why there is nothing in the index.